### PR TITLE
Cleanup / improvements

### PR DESCRIPTION
**Pull request by OrionNebula**
Added doccomments
Moved NAPI helpers to napi_helpers.h
Broke ts and js lint into two scripts
Changed to tslint-config-standard
Made @types/node a real dependency (necessary because we return / use types from that module in a public way)
Added export to index.d.ts
Changed "types" to "typings" and updated path

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[*.{ts,tsx}]
```
Comment (ashlynnlevenick) [inline .editorconfig]: This will likely stomp the existing .editorconfig ruleset, consider removing the file?

File: package.json [hunk 2]
```diff
@@ -3,26 +3,30 @@
   "version": "0.0.0",
   "description": "Decode arbitrary audio to PCM using FFmpeg",
   "main": "build/Release/binding.node",
-  "types": "types",
+  "typings": "types/index.d.ts",
```
Comment (ashlynnlevenick) [inline package.json]: I'd rather use the "types" name, as "typings" is just weird.

File: src/napi_helpers.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#ifndef __NAPI_HELPERS_H__
```
Comment (ashlynnlevenick) [inline src/napi_helpers.h]: Items prefixed with a double underscore are reserved for the compiler, please use `NAPI_HELPERS_H` instead.

File: types/index.d.ts [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
+   * @param whence The method for computing the final position from the offset
+   */
+  seek? (offset: number, whence: Whence): number
+  /** The total length of this stream, in bytes */
+  length? (): number
 }
 
-declare class Decoder {
-  constructor(subject: Decodeable);
+/** An object capable of decoding a stream using FFMPEG */
```
Comment (ashlynnlevenick) [inline types/index.d.ts]: The proper capitalization of FFmpeg's name is FFmpeg.

File: types/tsconfig.json [hunk 5]
```diff
@@ -1,11 +1,61 @@
...
   "compilerOptions": {
-    "module": "commonjs",
-    "lib": ["es6"],
-    "noImplicitAny": true,
-    "noImplicitThis": true,
-    "strictNullChecks": true,
-    "strictFunctionTypes": true,
-    "noEmit": true
+    /* Basic Options */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
```
Comment (ashlynnlevenick) [inline types/tsconfig.json]: This won't be compiled, I don't see the point.

File: types/tsconfig.json [hunk 6]
```diff
@@ -1,11 +1,61 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
-    "lib": ["es6"],
-    "noImplicitAny": true,
-    "noImplicitThis": true,
-    "strictNullChecks": true,
-    "strictFunctionTypes": true,
-    "noEmit": true
+    /* Basic Options */
```
Comment (ashlynnlevenick) [inline types/tsconfig.json]: Please refrain from adding comments that do not add to the context.

File: types/tsconfig.json [hunk 7]
```diff
@@ -1,11 +1,61 @@
...
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
```
Comment (ashlynnlevenick) [inline types/tsconfig.json]: Your IDE can tell you appropriate values for this field.

File: types/tsconfig.json [hunk 8]
```diff
@@ -1,11 +1,61 @@
...
+    "lib": [
+      "es6"
+    ],                             /* Specify library files to be included in the compilation. */
```
Comment (ashlynnlevenick) [inline types/tsconfig.json]: Missing newline?

File: types/tsconfig.json [hunk 9]
```diff
@@ -1,11 +1,61 @@
...
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
```
Comment (ashlynnlevenick) [inline types/tsconfig.json]: Why are all of these things in here but commented out.

File: types/tsconfig.json [hunk 10]
```diff
@@ -1,11 +1,61 @@
...
+
+    /* Module Resolution Options */
+    // "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+    // "baseUrl": "./",                       /* Base directory to resolve non-absolute module names. */
+    // "paths": {},                           /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+    // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
+    // "typeRoots": [],                       /* List of folders to include type definitions from. */
+    // "types": [],                           /* Type declaration files to be included in compilation. */
+    // "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+    "esModuleInterop": true,                  /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
```
Comment (ashlynnlevenick) [inline types/tsconfig.json]: Again, we aren't emitting.

**Review by lynlevenick — CHANGES_REQUESTED** [review]

File: types/index.d.ts [hunk 11]
```diff
@@ -1,15 +1,36 @@
-// Set from the beginning, offset from current, and set from end
-declare type Whence = 0 | 1 | 2;
+/** Set from the beginning, offset from current, and set from end */
+export declare type Whence = 0 | 1 | 2
```
Comment (ashlynnlevenick) [inline types/index.d.ts]: Per https://www.typescriptlang.org/docs/handbook/declaration-files/by-example.html , shouldn't this just be declare?

**Review by lynlevenick — CHANGES_REQUESTED** [review]

File: types/index.d.ts (hunk 11 above)
Comment (OrionNebula) [reply types/index.d.ts]: All declaration files generated by `tsc` use `export declare`

File: package.json [hunk 12]
```diff
@@ -3,26 +3,30 @@
...
     "node-gyp": "^3.7.0",
     "should": "^13.2.1",
     "standard": "^11.0.1",
     "tslint": "^5.11.0",
-    "tslint-eslint-rules": "^5.3.1",
+    "tslint-config-standard": "^7.1.0",
     "typescript": "^2.9.2"
+  },
+  "dependencies": {
+    "@types/node": "^10.5.2"
```
Comment (ashlynnlevenick) [inline package.json:30]: I'm not sure I see the necessity of this being a runtime dependency. It's making pure javascript consumers pull in a bunch of typescript stuff, which is also a bit grating.

File: types/index.d.ts (hunk 11 above)
Comment (ashlynnlevenick) [reply types/index.d.ts]: Is export declare necessary for typescript consumers to access the module?

File: package.json (hunk 12 above)
Comment (OrionNebula) [reply package.json:30]: See: https://stackoverflow.com/questions/45176661/how-do-i-decide-whether-types-goes-into-dependencies-or-devdependencies

Comment (ashlynnlevenick) [reply package.json:30]: Sounds good then.

File: types/index.d.ts (hunk 11 above)
Comment (OrionNebula) [reply types/index.d.ts]: I'm pretty sure, yeah

**Comment by lynlevenick** [comment]
Everything seems good, thank you for making those changes.

**Review by lynlevenick — APPROVED** [review]

**Pull request merged**